ember-cli/ember-cli
# Switch away from travis

TODO:
 - testing
   - `ember new` in a new repo, pushed to Github
      - [ ] with npm
      - [ ] with yarn
   - `ember addon` in a new repo, pushed to Github
     - [ ] with npm
     - [ ] with yarn
  - [ ] replace automated ember-cli tests that check for a .travis.yml file for checking for gh actions files
     

For Both:
 - split linting and tests in to separate jobs to maximize parallel running to try to best decrease over all CI time
 - support `[skip ci]` magic commit messages
 - use Volta
 
For Apps:
 - build once for tests and production
 - cache dependencies between jobs, because GH Action's cache is faster than re-installing (even for small monorepos)
   - it's always easier to remove the cache and switch to install on every job that to figure out how to get the cache working
   - this makes it easier for folks to scale and add things like netlify deploy and deploy preview without relying on external deploy infrastructure (which can have predatory policy changes / send you bills when you've been on the free tier) -- example deploy config (not included in this PR): https://github.com/NullVoxPopuli/limber/blob/main/.github/workflows/ci.yml#L190-L276
 
 For Addons:
  - run on a weekly cron so that maintained projects can keep an eye on what things break over time (if any)
